packages/sdk/src/reliable_channel/index.ts

```diff
@@ -0,0 +1,633 @@
+        this._retrieve,
+        async (msg: T) => {
+          await this.processIncomingMessage(msg);
+          await this.messageChannel.processTasks();
+          this.messageChannel.sweepIncomingBuffer();
+        }
+      );
+    }
+
+    this.isStarted = false;
+  }
+
+  /**
+   * Used to identify messages, pass the payload of a message you are
+   * about to send to track the events for this message.
+   * This is pre-sds wrapping
+   * @param messagePayload
+   */
+  public static getMessageId(messagePayload: Uint8Array): string {
+    return MessageChannel.getMessageId(messagePayload);
+  }
+
+  /**
+   * Create a new message channels. Message channels enables end-to-end
+   * reliability by ensuring that all messages in the channel are received
+   * by other users, and retrieved by this local node.
+   *
+   * emits events about outgoing messages, see [[`ReliableChannel`]] docs.
+   *
+   * Note that all participants in a message channels need to get the messages
+   * from the channel. Meaning:
+   * - all participants must be able to decrypt the messages
+   * - all participants must be subscribing to content topic(s) where the messages are sent
+   *
+   * @param node The waku node to use to send and receive messages
+   * @param channelId An id for the channel, all participants of the channel should use the same id
+   * @param senderId An id for the sender, to ensure acknowledgements are only valid if originating from someone else; best if persisted between sessions
+   * @param encoder A channel operates within a singular encryption layer, hence the same encoder is needed for all messages
+   * @param decoder A channel operates within a singular encryption layer, hence the same decoder is needed for all messages
+   * @param options
+   */
+  public static async create<T extends IDecodedMessage>(
+    node: IWaku,
+    channelId: ChannelId,
+    senderId: SenderId,
+    encoder: IEncoder,
+    decoder: IDecoder<T>,
+    options?: ReliableChannelOptions
+  ): Promise<ReliableChannel<T>> {
+    const sdsMessageChannel = new MessageChannel(channelId, senderId, options);
+    const messageChannel = new ReliableChannel(
+      node,
+      sdsMessageChannel,
+      encoder,
+      decoder,
+      options
+    );
+
+    const autoStart = options?.autoStart ?? true;
+    if (autoStart) {
+      await node.start();
+      await messageChannel.start();
+    }
+
+    return messageChannel;
+  }
+
+  /**
+   * Sends a message in the channel, will attempt to re-send if not acknowledged
+   * by other participants.
+   *
+   * @param messagePayload
+   */
+  public send(messagePayload: Uint8Array): void {
+    const messageId = ReliableChannel.getMessageId(messagePayload);
+    if (!this.isStarted) {
+      this.safeSendEvent(ReliableChannelEvent.OutMessageIrrecoverableError, {
+        detail: { messageId: messageId, error: "channel is not started" }
+      });
+    }
+    const send = this._wrapAndSend.bind(this, messagePayload);
```

Comment:
nit: `send` variable in `send` method doesn't look good, maybe `_send` variable name?